File: 88.merge-sorted-array/main.go
package main

func main() {

}

func merge(nums1 []int, m int, nums2 []int, n int) {
	for i, j, k := m-1, n-1, m+n-1; k >= 0; k-- {
		if i >= 0 && (j < 0 || nums1[i] >= nums2[j]) {
			nums1[k] = nums1[i]
			i--
		} else if j >= 0 && (i < 0 || nums2[j] >= nums1[i]) {
			nums1[k] = nums2[j]
			j--
		}
	}
}

File: 88.merge-sorted-array/main.rs
impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        let mut j = m - 1;
        let mut k = n - 1;
        for i in (0..nums1.len()).rev() {
            if j >= 0 && (k < 0 || nums1[j as usize] > nums2[k as usize]) {
                nums1[i] = nums1[j as usize];
                j -= 1;
            } else {
                nums1[i] = nums2[k as usize];
                k -= 1;
            }
        }
    }
}
